packages/cli/src/Server.ts

```diff
@@ -711,6 +754,64 @@ class App {
 			};
 		}));
 
+		// Retrieves all tags, with or without usage count
+		this.app.get(`/${this.restEndpoint}/tags`, ResponseHelper.send(async (req: express.Request, res: express.Response): Promise<ITagDb[] | Array<{ id: string, name: string, usageCount?: number }>> => {
+			if (req.query.withUsageCount === 'true') {
+				const foundTags = await TagHelpers.getAllTagsWithUsageCount();
+				return foundTags.map(({ id, name, usageCount }) => ({ id: id.toString(), name, usageCount }));
+			}
+
+			const foundTags = await Db.collections.Tag!.find({ select: ['id', 'name'] });
+			return foundTags.map(({ id, name }) => ({ id: id.toString(), name }));
+		}));
+
+		// Creates a tag
+		this.app.post(`/${this.restEndpoint}/tags`, ResponseHelper.send(async (req: express.Request, res: express.Response): Promise<{ id: string, name: string }> => {
+			TagHelpers.validateRequestBody(req.body);
+
+			const { name } = req.body;
+			await TagHelpers.validateName(name);
+			TagHelpers.validateLength(name);
```

Comment:
we can move all this validation under one function `validateName` since they all seem to be validating name property
```suggestion
			await TagHelpers.validateName(req.body);
```